crypto/ocsp/ocsp_test.cc

```diff
@@ -680,3 +688,54 @@ TEST_P(OCSPTest, VerifyOCSPResponse) {
   const int ocsp_verify_status = OCSP_basic_verify(basic_response.get(), server_cert_chain.get(), trust_store.get(), 0);
   ASSERT_EQ(t.expected_ocsp_verify_status, ocsp_verify_status);
 }
+
+struct OCSPReqTestVector {
+  const char *ocsp_request;
+  int expected_status;
+};
+
+static const OCSPReqTestVector kRequestTestVectors[] = {
+    {"ocsp_request", OCSP_RESQUEST_PARSE_SUCCESS},
+    {"ocsp_request_attached_cert", OCSP_RESQUEST_PARSE_SUCCESS},
+    {"ocsp_request_no_nonce", OCSP_RESQUEST_PARSE_SUCCESS},
+    {"ocsp_request_signed", OCSP_RESQUEST_PARSE_SUCCESS},
+    {"ocsp_request_signed_sha256", OCSP_RESQUEST_PARSE_SUCCESS},
+    {"ocsp_response", OCSP_RESQUEST_PARSE_ERROR},
+};
+
+class OCSPReqParseTest : public testing::TestWithParam<OCSPReqTestVector> {};
+
+INSTANTIATE_TEST_SUITE_P(All, OCSPReqParseTest,
+                         testing::ValuesIn(kRequestTestVectors));
+
+TEST_P(OCSPReqParseTest, OCSPRequestParse) {
+   const OCSPReqTestVector &t = GetParam();
+
+  std::string data = GetTestData(std::string("crypto/ocsp/test/aws/" +
+                                std::string(t.ocsp_request) + ".der").c_str());
+  std::vector<uint8_t> ocsp_request_data(data.begin(), data.end());
+
+  bssl::UniquePtr<BIO> bio(BIO_new(BIO_s_mem()));
+  bssl::UniquePtr<OCSP_REQUEST> ocspRequest =
+      LoadOCSP_REQUEST(ocsp_request_data);
+
+  if(t.expected_status) {
+    ASSERT_TRUE(ocspRequest);
+    // If request parsing is successful, try setting up a |OCSP_REQ_CTX| with
+    // default settings.
+    bssl::UniquePtr<OCSP_REQ_CTX> ocspReqCtx(OCSP_sendreq_new(bio.get(),
+                                                nullptr, ocspRequest.get(), 0));
+    ASSERT_TRUE(ocspReqCtx);
+
+    // Set up |OCSP_REQ_CTX| without a |OCSP_REQUEST|. Then finalize the context
+    // later.
+    bssl::UniquePtr<OCSP_REQ_CTX> ocspReqCtxLater(OCSP_sendreq_new(bio.get(),
+                                                nullptr, nullptr, 0));
+    ASSERT_TRUE(ocspReqCtx);
+    ASSERT_TRUE(OCSP_REQ_CTX_set1_req(ocspReqCtxLater.get(),
+                                      ocspRequest.get()));
```

Comment:
Is it possible to test that this finalized request context is finalized in the way we expect? Or that the content is sane?